src/accelerator.rs

```diff
@@ -0,0 +1,281 @@
+//! of an optional combination of modifier keys (provided by [`SysMods`](crate::accelerator::SysMods),
+//! [`RawMods`](crate::accelerator::RawMods) or [`Modifiers`](crate::accelerator::Modifiers)) and
+//! one key ([`Code`](crate::accelerator::Code)).
+//!
+//! # Examples
+//! They can be created directly
+//! ```
+//! # use muda::accelerator::{Accelerator, Mods, Modifiers, Code};
+//! #
+//! let accelerator = Accelerator::new(Mods::Shift, Code::KeyQ);
+//! let accelerator_with_raw_mods = Accelerator::new(Mods::Shift, Code::KeyQ);
+//! let accelerator_without_mods = Accelerator::new(None, Code::KeyQ);
+//! # assert_eq!(accelerator, accelerator_with_raw_mods);
+//! ```
+//! or from `&str`, note that all modifiers
+//! have to be listed before the non-modifier key, `shift+alt+q` is legal,
+//! whereas `shift+q+alt` is not.
+//! ```
+//! # use muda::accelerator::{Accelerator, Mods};
+//! #
+//! let accelerator: Accelerator = "shift+alt+q".parse().unwrap();
+//! #
+//! # // This assert exists to ensure a test breaks once the
+//! # // statement above about ordering is no longer valid.
+//! # assert!("shift+q+alt".parse::<Accelerator>().is_err());
+//! ```
+//!
+
+pub use keyboard_types::{Code, Modifiers};
+use std::{borrow::Borrow, hash::Hash, str::FromStr};
+
+/// Base `Accelerator` functions.
+#[derive(Debug, Clone, PartialEq, Hash)]
+pub struct Accelerator {
+    pub(crate) mods: Modifiers,
+    pub(crate) key: Code,
+}
+
+impl Accelerator {
+    /// Creates a new accelerator to define keyboard shortcuts throughout your application.
+    pub fn new(mods: impl Into<Option<Modifiers>>, key: Code) -> Self {
+        Self {
+            mods: mods.into().unwrap_or_else(Modifiers::empty),
+            key,
+        }
+    }
+
+    /// Returns `true` if this [`Code`] and [`Modifiers`] matches this `Accelerator`.
+    ///
+    /// [`Code`]: Code
+    /// [`Modifiers`]: crate::accelerator::Modifiers
+    pub fn matches(&self, modifiers: impl Borrow<Modifiers>, key: impl Borrow<Code>) -> bool {
+        // Should be a const but const bit_or doesn't work here.
+        let base_mods = Modifiers::SHIFT | Modifiers::CONTROL | Modifiers::ALT | Modifiers::SUPER;
+        let modifiers = modifiers.borrow();
+        let key = key.borrow();
+        self.mods == *modifiers & base_mods && self.key == *key
+    }
+}
+
+// Accelerator::from_str is available to be backward
+// compatible with tauri and it also open the option
+// to generate accelerator from string
+impl FromStr for Accelerator {
+    type Err = AcceleratorParseError;
+    fn from_str(accelerator_string: &str) -> Result<Self, Self::Err> {
+        parse_accelerator(accelerator_string)
+    }
+}
+
+/// Represents the active modifier keys.
+///
+/// This is intended to be clearer than [`Modifiers`], when describing accelerators.
+///
+#[non_exhaustive]
+#[derive(Debug, Clone, Copy, PartialEq, Hash)]
+pub enum Mods {
+    None,
+    Alt,
+    Ctrl,
```

Comment:
Missing variants for macos
```suggestion
    Ctrl,
	Command,
    CommandOrCtrl,
```